Client/Frontend/Browser/TabManagerNavDelegate.swift

```diff
@@ -41,18 +41,19 @@ class TabManagerNavDelegate: NSObject, WKNavigationDelegate {
     }
   }
 
-  func webView(_ webView: WKWebView, didReceive challenge: URLAuthenticationChallenge, completionHandler: @escaping (URLSession.AuthChallengeDisposition, URLCredential?) -> Void) {
+  @MainActor
+  public func webView(_ webView: WKWebView, respondTo challenge: URLAuthenticationChallenge) async -> (URLSession.AuthChallengeDisposition, URLCredential?) {
     let authenticatingDelegates = delegates.filter { wv in
-      return wv.responds(to: #selector(webView(_:didReceive:completionHandler:)))
+      return wv.responds(to: #selector(WKNavigationDelegate.webView(_:didReceive:completionHandler:)))
     }
 
     guard let firstAuthenticatingDelegate = authenticatingDelegates.first else {
-      return completionHandler(.performDefaultHandling, nil)
+      return (.performDefaultHandling, nil)
     }
 
-    firstAuthenticatingDelegate.webView?(webView, didReceive: challenge) { (disposition, credential) in
-      completionHandler(disposition, credential)
-    }
+    // Do NOT change to `delegate.webView?(....)` the optional operator makes async-await calls crash the compiler atm!
+    // It must be force-unwrapped at the time of writing `January 17th, 2023`.
```

Comment:
this comment still valid?

Reply:
Still valid as of January 17th. We have this comment on all other WebKit async delegate methods as well